app/assets/javascripts/discourse/tests/integration/components/header-contents-test.gjs

```diff
@@ -8,24 +8,82 @@ import { setupRenderingTest } from "discourse/tests/helpers/component-test";
 module("Integration | Component | Header | Contents", function (hooks) {
   setupRenderingTest(hooks);
 
-  test("showHeaderSearch", async function (assert) {
-    const site = getOwner(this).lookup("service:site");
-    const toggleNavigationMenu = () => {};
-
-    sinon.stub(site, "mobileView").value(true);
-
-    await render(
-      <template>
-        <Contents
-          @sidebarEnabled={{true}}
-          @toggleNavigationMenu={{toggleNavigationMenu}}
-          @showSidebar={{true}}
-        >test</Contents>
-      </template>
-    );
-
-    assert
-      .dom(".floating-search-input-wrapper")
-      .doesNotExist("it does not display when the site is in mobile view");
+  module("header search", function () {
+    test("is hidden in mobile view", async function (assert) {
+      const site = getOwner(this).lookup("service:site");
+      const toggleNavigationMenu = () => {};
+
+      sinon.stub(site, "mobileView").value(false);
+
+      await render(
+        <template>
+          <Contents
+            @sidebarEnabled={{true}}
+            @toggleNavigationMenu={{toggleNavigationMenu}}
+            @showSidebar={{true}}
+          >
+            test
+          </Contents>
+        </template>
+      );
+
+      assert
+        .dom(".floating-search-input-wrapper")
+        .doesNotExist("it does not display when the site is in mobile view");
+    });
+
+    ["signup", "login", "invites.show", "activate-account"].forEach((name) => {
+      test(`is hidden in route "${name}"`, async function (assert) {
+        const router = getOwner(this).lookup("service:router");
+        const toggleNavigationMenu = () => {};
+
+        sinon.stub(router, "currentRouteName").value(name);
+
+        await render(
+          <template>
+            <Contents
+              @sidebarEnabled={{true}}
+              @toggleNavigationMenu={{toggleNavigationMenu}}
+              @showSidebar={{true}}
```

Comment:
are all three arguments required in these tests or can they be omitted?

Reply:
Thanks, @CvX! All three are unnecessary, as it turned out 😄